network/napalm/napalm_install_config.py

```diff
@@ -0,0 +1,226 @@
+description:
+    - "This library will take the configuration from a file and load it into a device running any OS supported by napalm.
+      The old configuration will be replaced or merged with the new one."
+requirements:
+    - napalm
+options:
+    hostname:
+        description:
+          - IP or FQDN of the device you want to connect to
+        required: True
+    username:
+        description:
+          - Username
+        required: True
+    password:
+        description:
+          - Password
+        required: True
+    dev_os:
+        description:
+          - OS of the device
+        required: True
+        choices: ['eos', 'junos', 'iosxr', 'fortios', 'ibm', 'ios', 'nxos']
+    timeout:
+        description:
+          - Time in seconds to wait for the device to respond
+        required: False
+        default: 60
+    optional_args:
+        description:
+          - Dictionary of additional arguments passed to underlying driver
+        required: False
+    config_file:
+        description:
+          - Path to the file to load the configuration from
+        required: True
+    commit_changes:
+        description:
+          - If set to True the configuration will be actually merged or replaced. If the set to False,
+            we will not apply the changes, just check and report the diff
+        required: True
+    replace_config:
+        description:
+          - If set to True, the entire configuration on the device will be replaced during the commit.
+            If set to False, we will merge the new config with the existing one. Default- False
+        choices: [yes,on,1,true,no,off,0,false]
+        default: False
+        required: False
+    diff_file:
+        description:
+          - A path to the file where we store the "diff" between the running configuration and the new
+            configuration. If not set the diff between configurations will not be saved.
+        default: None
+        required: False
+    get_diffs:
+        description:
+            - Set to False to not have any diffs generated. Useful if platform does not support commands
+              being used to generate diffs. Note- By default diffs are generated even if the diff_file
+              param is not set.
+        choices: [yes,on,1,true,no,off,0,false]
+        default: True
+        required: False
+'''
+
+EXAMPLES = '''
+- assemble:
+    src=../compiled/{{ inventory_hostname }}/
+    dest=../compiled/{{ inventory_hostname }}/running.conf
+
+ - napalm_install_config:
+    hostname={{ inventory_hostname }}
+    username={{ user }}
+    dev_os={{ os }}
+    password={{ passwd }}
+    config_file=../compiled/{{ inventory_hostname }}/running.conf
+    commit_changes={{ commit_changes }}
+    replace_config={{ replace_config }}
+    get_diffs=True
+    diff_file=../compiled/{{ inventory_hostname }}/diff
+'''
+
+RETURN = '''
+changed:
```

Comment:
Is this not needed because it is not a usual ansible return key? I figured we'd document it as we do return "changed":
    module.exit_json(changed=changed, msg=diff)